test/ci/verify-templates.test.js

```diff
@@ -0,0 +1,137 @@
+    const projectRoot = path.resolve(__dirname, '../..');
+    // Assemble a list of template directories here, since templates.json
+    // may not have all of them:
+    const templates = fs
+          .readdirSync(projectRoot, { withFileTypes: true })
+          .filter(
+              (file) =>
+              file.isDirectory() &&
+                  !file.name.startsWith('.') &&
+                  !file.name.startsWith('_') &&
+                  !excludedPaths.includes(file.name)
+          )
+          .map((dir) => dir.name);
+
+    describe('CI template verification', () => {
+        describe.each(templates)('the "%s" function template', (template) => {
+            it('should have a populated functions directory', (done) => {
+                const functionsDir = path.join(projectRoot, template,
+                                               'functions');
+
+                fs.readdir(functionsDir, (err, files) => {
+                    expect(err).toBeFalsy();
+                    expect(files.length).toBeGreaterThan(0);
+                    done();
+                });
+            });
+
+            it('should have a populated tests directory', (done) => {
+                const functionsDir = path.join(projectRoot, template, 'tests');
+
+                fs.readdir(functionsDir, (err, files) => {
+                    expect(err).toBeFalsy();
+                    expect(files.length).toBeGreaterThan(0);
+                    done();
+                });
+            });
+
+            describe('its assets/index.html file', () => {
+                const indexFile = path.join(projectRoot, template,
+                                            'assets', 'index.html');
+
+                it('should exist', (done) => {
+                    fs.access(indexFile, fs.constants.F_OK, (err) => {
+                        expect(err).toBeFalsy();
+                        done();
+                    });
+                });
+
+                it('should contain <!-- APP_INFO_V2 --> (for new-style index pages)', (done) => {
+                    fs.readFile(indexFile, (err, data) => {
+                        expect(err).toBeFalsy();
+                        if(data.includes('ce-paste-theme.css')) {
+                            expect(data.includes('<!-- APP_INFO_V2 -->'));
+                        }
+                        done();
+                    });
+                });
+            });
+
+            describe('its .env file', () => {
+                const envFile = path.join(projectRoot, template, '.env');
+
+                it('should exist', (done) => {
+                    fs.access(envFile, fs.constants.F_OK, (err) => {
+                        expect(err).toBeFalsy();
+                        done();
+                    });
+                });
+
+                it('should be parseable', async () => {
+                    const result = parser.parseFile(envFile);
```

Comment:
should this be `const result = await parser.parseFile(envFile);` and then remove the `await` before the `expect`

Reply:
They're synonymous, but I changed it to your suggestion for consistency with the other `.env` tests.